tangos/input_handlers/finding.py

```diff
@@ -48,14 +48,23 @@ def best_matching_handler(cls, basename):
         base = os.path.join(config.base, basename)
         if len(cls.__subclasses__()) == 0:
             return cls
-        all_possible_handlers = cls.__subclasses__()
+
+        # Add all subclasses and sub-subclasses
+        all_possible_handlers = []
```

Comment:
This might cause confusion if people make a sub-sub-sub-class (not impossible).

I think it would be better to move this out into a utility function called `get_all_subclasses` or something, and make it recursive so that it really captures _all_ subclasses.

Reply:
I agree it looks like the logic would be better isolated somewhere else, and made recursive. @cphyc could you attempt this? I haven't touched this part of the PR at all.

Reply:
At the moment, all (sub-)*classes are being added, even though the code isn't recursive (it uses a stack of classes and subclasses until the stack is exhausted, adding to the stack at each iteration the current class' subclasses). I've updated the comment accordingly.

Reply:
Ah yes. Would it still be a good idea to put it in a separate function though?

Reply:
@apontzen done!